### PR TITLE
feat(cosmos): replace StakingOpportunities -> EarnOpportunities

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Discussed this one out with @reallybeard, very easy lift to fix the default validator regression for accounts without an active Cosmos SDK staking opportunity, and makes things properly abstracted without a Cosmos-SDK-specific component.

Note: Account page granularity is broken and we never wrote any logic for it - it gives us the aggregated amounts. This isn't Cosmos SDK specific and will be fixed in a follow-up.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- closes https://github.com/shapeshift/web/issues/3607

## Risk

This doesn't show rewards anymore, though this is now consistent. We can add them for *all* earn opportunities as a follow-up, instead of it just being present for Cosmos SDK.

## Testing

- Cosmos SDK account/asset pages staking opportunities are happy
- Test both with a wallet with active opportunities, as well as with a wallet with no ATOM/OSMO active opportunities to ensure the default validator is present

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

#### Asset page

image

#### Account page

image

#### Asset page

image

### Account page - not OK, we should not aggregate, but not a regression either

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3840** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3840" target="_blank">Graphite</a> 
    * **PR #3839** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3839" target="_blank">Graphite</a> 
      * **PR #3841** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3841" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3841?utm_source=stack-comment).

File: src/plugins/cosmos/CosmosAssetAccountDetails.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const CosmosAssetAccountDetails = ({ assetId, chainId, accountId }: Asset
           {accountId && <AccountAssets assetId={assetId} accountId={accountId} />}
           <AssetAccounts assetId={assetId} accountId={accountId} />
           {supportsStaking(fromAssetId(assetId).chainId) && (
-            <StakingOpportunities accountId={accountId} assetId={assetId} />
+            <EarnOpportunities assetId={assetId} accountId={accountId} />
```
Comment (gomesalexandre) [inline src/plugins/cosmos/CosmosAssetAccountDetails.tsx:47]: Prettier diff
```suggestion
            <EarnOpportunities accountId={accountId} assetId={assetId} />
```

File: src/components/StakingVaults/EarnOpportunities.tsx [hunk 2]
```diff
@@ -38,7 +38,9 @@ export const EarnOpportunities = ({ assetId, accountId }: EarnOpportunitiesProps
   const asset = useAppSelector(state => selectAssetById(state, assetId))
   if (!asset) throw new Error(`Asset not found for AssetId ${assetId}`)
 
-  const stakingOpportunities = useAppSelector(selectAggregatedEarnUserStakingOpportunities)
+  const stakingOpportunities = useAppSelector(
+    selectAggregatedEarnUserStakingOpportunitiesIncludeEmpty,
```
Comment (gomesalexandre) [inline src/components/StakingVaults/EarnOpportunities.tsx:42]: Fixes the default validator not being present in case you're not staked to it already

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by gomesalexandre** [comment]
Big @reallybeard approval ✅

**Pull request merged**